libgnucash/app-utils/gnc-gsettings.cpp

```diff
@@ -672,6 +672,23 @@ gnc_gsettings_get_user_value (const gchar *schema,
 
 using opt_str_vec = boost::optional<std::string>;
 
+static void
+clear_gset_object (const opt_str_vec &path)
+{
+    GSettings *gset = nullptr;
+
+    // Now release the gsettings object
+    gset = static_cast<GSettings*> (g_hash_table_lookup (schema_hash, path->c_str()));
+    DEBUG ("Looking for schema %s to release gsettings, returned %p",  path->c_str(), gset);
+
+    if (gset)
+    {
+        DEBUG ("Clearing gsettings object %p", gset);
+        g_hash_table_remove (schema_hash, path->c_str());
+        g_clear_object (&gset);
```

Comment:
g_clear_object doesn't add much here over g_object_unref.  gset will go out of scope right after this call, so there nowhere you'll ever test if the pointer gset itself is null. The object it points to is not affected by g_clear_object other than the usual refcounting machinery. If, when you call this function, there's only one reference left, the object will be freed from memory, otherwise its refcount will only be decreased. That's exactly the same as g_object_unref.

Reply:
Other than that, I don't think we'll need this at all. IMO we can just stop adding most objects in our hash table.

Reply:
OK, a bit over the top and as pointed out you PR replaces this commit so can be dropped.